cfgov/unprocessed/js/modules/post-filter.js

```diff
@@ -102,14 +102,30 @@ PostFilter.prototype = {
   },
 
   onSubmit: function( e ) {
-    var dateRange =
-      dateRangeFormatter.format( this.$gte.val(), this.$lte.val() );
-    if ( dateRange && dateRange.isValid ) {
+    var $form = $( e.target );
+    var fromDateValue = this.$gte.val()
+    var toDateValue = this.$lte.val()
+    var dateRange = dateRangeFormatter.format( fromDateValue, toDateValue );
+
+    if ( ( dateRange && dateRange.isValid ) ) {
       this.$gte.val( dateRange.startDate );
       this.$lte.val( dateRange.endDate );
     } else {
       e.preventDefault();
-      // TODO: Add inline notifcation;
+      $form.trigger( 'cf_notifier:notify', {
+            message: 'Please enter a valid Date Range.',
+            state:   'error'
+      } );
+    }
+
+    if( $form.hasClass( 'js-validate_require-date' ) &&
+        toDateValue == '' && fromDateValue == '') {
+      e.preventDefault();
```

Comment:
Fixed in https://github.com/cfpb/cfgov-refresh/commit/606b7c1d369eb210b18ded6cb7f660ffbe3bf87e